File: .cpp/442.FindAllDuplicatesInAnArray.cpp
class Solution
{
public:
    vector<int> findDuplicates(vector<int> &nums)
    {
        int n = nums.size();
        int i = 0;
        bool check = false;
        int value = 0;
        while (i < n)
        {
            value = nums[i] - 1;
            check = value >= 0 && value < n;
            if (check && nums[i] != nums[value])
            {
                swap(nums[value], nums[i]);
            }
            else
            {
                i++;
            }
        }
        vector<int> result;
        for (i = 0; i < n; i++)
        {
            if (nums[i] != i + 1)
            {
                result.push_back(nums[i]);
            }
        }
        return result;
    }
};

File: .py/442.FindAllDuplicatesInAnArray.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        n = len(nums)
        i = 0
        while i<n:
            value = nums[i] - 1
            check = value >=0 and value < n
            if check and nums[value]!=nums[i]:
                nums[value],nums[i] = nums[i],nums[value]
            else:
                i+=1
        result = []
        for i in range(n):
            if nums[i]!= i + 1:
                result.append(nums[i])
        return result